File: solutions/_535_encode_and_decode_tinyurl/solution.py
import random
import string


class Codec:
    def __init__(self):
        self.long2short = {}
        self.short2long = {}
        self.base = "http://tinyurl.com/"

    def encode(self, longUrl: str) -> str:
        if longUrl not in self.long2short:
            shortUrl = self.base + "".join(
                random.choice(string.ascii_lowercase) for _ in range(6)
            )
            self.long2short[longUrl] = shortUrl
            self.short2long[shortUrl] = longUrl
        return self.long2short[longUrl]

    def decode(self, shortUrl: str) -> str:
        return self.short2long[shortUrl]


# https://leetcode.com/problems/encode-and-decode-tinyurl/

File: solutions/_535_encode_and_decode_tinyurl/solution.js
const base = "http://tinyurl.com/";
const { randomBytes } = require("crypto");

const long2short = new Map();
const short2long = new Map();

var encode = function (longUrl) {
  if (!long2short.has(longUrl)) {
    const shortUrl = base + randomBytes(4).toString("hex");
    long2short.set(longUrl, shortUrl);
    short2long.set(shortUrl, longUrl);
  }
  return long2short.get(longUrl);
};

var decode = function (shortUrl) {
  return short2long.get(shortUrl);
};

module.exports = { encode, decode };

/*
https://leetcode.com/problems/encode-and-decode-tinyurl/
*/
